components/director/pkg/operation/api.go

```diff
@@ -0,0 +1,148 @@
+
+	"github.com/kyma-incubator/compass/components/director/pkg/resource"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/log"
+)
+
+const ResourceIDParam = "resource_id"
+const ResourceTypeParam = "resource_type"
+
+// ResourceFetcherFunc defines a function which fetches a particular resource by tenant and resource ID
+type ResourceFetcherFunc func(ctx context.Context, tenantID, resourceID string) (model.Entity, error)
+
+// TenantLoaderFunc defines a function which fetches the tenant for a particular request
+type TenantLoaderFunc func(ctx context.Context) (string, error)
+
+type handler struct {
+	transact            persistence.Transactioner
+	resourceFetcherFunc ResourceFetcherFunc
+	tenantLoaderFunc    TenantLoaderFunc
+}
+
+// NewHandler creates a new handler struct associated with the Operations API
+func NewHandler(transact persistence.Transactioner, resourceFetcherFunc ResourceFetcherFunc, tenantLoaderFunc TenantLoaderFunc) *handler {
+	return &handler{
+		transact:            transact,
+		resourceFetcherFunc: resourceFetcherFunc,
+		tenantLoaderFunc:    tenantLoaderFunc,
+	}
+}
+
+// ServeHTTP handles the Operations API requests
+func (h *handler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
+	ctx := request.Context()
+
+	tenantID, err := h.tenantLoaderFunc(ctx)
+	if err != nil {
+		log.C(ctx).WithError(err).Errorf("An error occurred while retrieving tenant from context: %s", err.Error())
+		apperrors.WriteAppError(ctx, writer, apperrors.NewInternalError("Unable to determine tenant for request"), http.StatusInternalServerError)
+		return
+	}
+
+	queryParams := request.URL.Query()
+
+	op := &Operation{
+		ResourceID:   queryParams.Get(ResourceIDParam),
+		ResourceType: resource.Type(queryParams.Get(ResourceTypeParam)),
+	}
+
+	log.C(ctx).Infof("Executing Operation API with resourceType: %s and resourceID: %s", op.ResourceType, op.ResourceID)
+
+	if err := op.Validate(); err != nil {
+		apperrors.WriteAppError(ctx, writer, apperrors.NewInvalidDataError("Unexpected resource type and/or GUID"), http.StatusBadRequest)
+		return
+	}
+
+	tx, err := h.transact.Begin()
+	if err != nil {
+		log.C(ctx).WithError(err).Errorf("An error occurred while opening db transaction: %s", err.Error())
+		apperrors.WriteAppError(ctx, writer, apperrors.NewInternalError("Unable to establish connection with database"), http.StatusInternalServerError)
+		return
+	}
+	defer h.transact.RollbackUnlessCommitted(ctx, tx)
+
+	ctx = persistence.SaveToContext(ctx, tx)
+
+	res, err := h.resourceFetcherFunc(ctx, tenantID, op.ResourceID)
+	if err != nil {
+		log.C(ctx).WithError(err).Errorf("An error occurred while fetching resource from database: %s", err.Error())
+
+		if apperrors.IsNotFoundError(err) {
+			apperrors.WriteAppError(ctx, writer, apperrors.NewNotFoundErrorWithMessage(resource.Application, op.ResourceID,
+				fmt.Sprintf("Operation for application with id %s not found", op.ResourceID)), http.StatusNotFound)
```

Comment:
some places it's `resource`, others - `application`. I guess because it's hardcoded for now, but we should have it in mind when introducing other types